### ISSUE TITLE
Parsing Error: Invalid regular expression with in non-bmp character set

### ISSUE THREAD

**Issue by preco21** [babel/babel-eslint#227]
When using RegExp with non-bmp character set like this:

```javascript
// case 1
string.replace(/[\u{0000A0}-\u{10FFFF}<>\&]/gmiu, (char) => `&#x${char.codePointAt(0).toString(16)};`); // error

// case 2
/[💫-💬]/u.test('💫'); // error

// case 3
/[💫💬]/u.test('💫'); // this works fine
```

then linter says:

```
Parsing error: Invalid regular expression: //[\u{0000A0}-\u{10FFFF}<>\&]/gmiu/: Range out of order in character classat line 1 col 1
```

Using modules:

```json
{
"babel-eslint": "^5.0.0-beta6",
"babel-plugin-add-module-exports": "^0.1.2",
"babel-preset-es2015": "^6.3.13",
"eslint": "^2.0.0-alpha-2",
"eslint-plugin-babel": "^3.0.0"
}
```

**Comment by hzoo** [comment]
Should be fixed with https://github.com/babel/acorn-to-esprima/commit/b55160f705da59314b1a6bed6395143a41b80030 - acorn-to-esprima 2.0.6 (dependency in babel-eslint) so I would just npm update/install again.
